docs/BASEIMAGE-UPDATE.md

```diff
@@ -0,0 +1,22 @@
+# Baseimage update
+
+The API and EVE code supports updating the base image (the EVE rootfs including the hypervisor, dom0 kernel, and the EVE agents/microservices) using dual partitions and priority boot support in grub. This is robust against getting a power failure during the write to the flash, and hung boot, as well as bugs in the EVE code or the dom0 kernel resulting in crashing or hung agents or e.g., a busted kernel Ethernet device driver which can no longer communicate.
+
+The EVE code current makes sure the device remains functional (can connect to the controller and no agent crashes or hangs) for 10 minutes running the new version, and then the system commits to that new version. Plan is to add a handshake with the controller so that the user can run their own application tests before committing to the new version.
+
+The timer behavior can be controlled by some variables specified in [global-config-variables.md](../pkg/pillar/docs/global-config-variables.md)
+
```

Comment:
Honestly this is the level of documentation I'd love to have: https://www.chromium.org/chromium-os/chromiumos-design-docs/disk-format and more specifically https://www.chromium.org/chromium-os/chromiumos-design-docs/disk-format#TOC-Selecting-the-kernel

Do you think this would be a reasonable expectation @nordmark ?

Reply:
I thought we had the partitions documented somewhere else.
In any case, I think that belongs in a separate file in /docs.

There is no notion of update and rollback in the device API (we have them in the cloud/controller API and in the UI and zcli but that has nothing to do with the device API)

The device API is "tell me an image and whether or not Activate is true" and "report back the current state for one or more images".

Note that "tell me an image", whether for app instance or baseos, includes specifying a datastore, signatures, etc in the device API. Not unique to the baseimage specification. Having sample code for the controller side of the API would make that more clear.

I'll create a separate issue for the partition format.

Reply:
@nordmark I guess what I'm saying is that we need an overall doc like what Chrome has that answers the question of how BaseOS functions. Partitioning and partition bits and how zboot works, etc is all part of those implementation details but I think it needs to be pulled into the same doc.

Specifically for device API: I think the above few paragraphs is exactly what needs to be added to the doc ;-) Basically saying "here's what device does (super simple) and it is simple because here's what we expect controller to do"

So what I would suggest at this point is to merge this PR and work on an overall doc a'la Chrome one. Will this make sense?

Reply:
We do need that gpt/grub part of the description (but I have no idea about the details of where the priority bit lives in the GPT - @gianlzed would know.)

Also, we should add information about the partition layout. Some of that is covered in the build description, but we should also have that description independently of the build.

Let's add those to the list of additional things to document, and merge this PR once I fix the issues below.